File: EpersBackend/Epers.Models/Nomenclatoare/NObiective.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Epers.Models.Nomenclatoare
{
	public class NObiective
	{
        [Key]
        public int Id { get; set; }

        [Required]
        public string Denumire { get; set; } = string.Empty;

        public int? IdFirma { get; set; }
        public int? IdCompartiment { get; set; }
        public int? IdPost { get; set; }
        public decimal? ValMin { get; set; }
        public decimal? ValTarget { get; set; }
        public decimal? ValMax { get; set; }
        public decimal? BonusMin { get; set; }
        public decimal? BonusTarget { get; set; }
        public decimal? BonusMax { get; set; }
        public string Frecventa { get; set; } = string.Empty;
        public decimal? Pondere { get; set; }
        public bool IsFaraProcent { get; set; }
        public string Tip { get; set; } = string.Empty;
        public string UpdateId { get; set; } = string.Empty;
        public DateTime? UpdateDate { get; set; }
        public bool? IsBonusProcentual { get; set; }

        [NotMapped]
        public string Firma { get; set; } = string.Empty;
    }
}

File: epers-front-end/src/app/models/nomenclatoare/NObiective.ts
export interface NObiective {
    id: number;
    denumire: string;
    idFirma: number | null;
    idCompartiment: number | null;
    idPost: number | null;
    valMin: number | null;
    valTarget: number | null;
    valMax: number | null;
    bonusMin: number | null;
    bonusTarget: number | null;
    bonusMax: number | null;
    frecventa: string;
    firma: string;
    pondere: number | null;
    isFaraProcent: boolean;
    tip: string;
    updateId: string;
    updateDate: Date | null;
    isBonusProcentual: boolean | null;
}

export interface NObiectiveDisplayModel {
    nomObiectiveData: NObiective[];
    pages: number;
    currentPage: number;
}


export interface AfisareNObiective {
    id: number;
    post: string;
    compartiment: string;
    denumire: string;
    tip: string;
    valTarget: number | null;
    valMin: number | null;
    valMax: number | null;
    bonusTarget: number | null;
    bonusMin: number | null;
    bonusMax: number | null;
    frecventa: string | null;
    isFaraProcent: boolean;
    firma: string;
    idFirma: number | null;
}

export interface AfisareNObiectiveDisplayModel {
    afisNomObiectiveData: AfisareNObiective[];
    pages: number;
    currentPage: number;
}